File: python/easy/shortest_unsorted_continuous_subarray.py
class Solution(object):
    def findUnsortedSubarray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return 0;
        sorted_array = sorted(nums)
        if nums == sorted_array:
            return 0
        i = min(i for i in range(len(nums)) if nums[i] != sorted_array[i])
        j = max(i for i in range(len(nums)) if nums[i] != sorted_array[i])
        return j-i+1
        

File: cpp/easy/shortest_unsorted_continuous_subarray.cpp
class Solution {
public:
    int findUnsortedSubarray(vector<int>& nums) {
        if(nums.size() <= 1)
            return 0;
        int start = -1;
        int len = 0;
        for(int i = 1; i < nums.size(); ++i)
        {
            int j = 0;
            if(nums[i] < nums[i-1])
            {
                j = i;
                while(j > 0 && nums[j] < nums[j-1])
                {
                    swap(nums[j], nums[j-1]);
                    j--;
                }
                if(start == -1 || start > j)
                    start = j;
                len = max(len, i -start + 1);
            }
            
        }
        return len;
    }
};

Conclusion:
compare every two adjacent element in the array nums[i] and nums[i-1], if nums[i-1] > nums[i], then swap the two elements, shift the 
large elements until it is less than the larger value. Them recaculate the len with i-start+1.
Time Complexity: O(n2) worst case
Space Complexity: O(1)
